doc/release_process.plantuml

```diff
@@ -0,0 +1,53 @@
+@startuml
```

Comment:
This diagram is rendered in https://github.com/wheleph/nttt/wiki/Developing-workflow

Reply:
An alternative approach could be:

- After pre-release _a.b.c_ has been created, create a test branch (e.g. _test_a_b_c_) from _improvements_.
- Customer testing is done in _test_a_b_c_. Bugs found during testing can be fixed in this branch, but also have to be merged to _improvements_.
- Development of version _x.y.z_ can start in _improvements_.
- Once customer testing has finished, all bugs have been fixed, verified and merged, version _a.b.c_ can be released and merged to _stable_.

Advantages: Version _a.b.c_ can be released, even when bugs were found during customer testing. Development of version _x.y.z_ can start while customer testing is ongoing.
Disadvantage: Bug fixes have to be merged from the test branch to _improvements_.

Reply:
> Version a.b.c can be released, even when bugs were found during customer testing.

But to achieve this we need to move the tag _a.b.c_ to the revision where all bugs found are fixed, shouldn't we? I'd like to keep tags immutable to make things more traceable.

> Development of version x.y.z can start while customer testing is ongoing.

I hope that customer testing phase will not be that long. And in case if it is we can always create a temporary branch _further-improvements_ for our work.

Also I propose to name the upcoming release `0.3.0` because in `master` the (broken) version is `0.2.1`. This approach will make obvious that our changes are the most up-to-date.

What do you think @robertjank ?

Reply:
> But to achieve this we need to move the tag a.b.c to the revision where all bugs found are fixed, shouldn't we? I'd like to keep tags immutable to make things more traceable.
We could tag it as a.b.c_RC (release candidate) before customer testing starts and a.b.c when customer testing is finished and all bugs fixed.

> I hope that customer testing phase will not be that long. And in case if it is we can always create a temporary branch further-improvements for our work.
Agreed. Maybe we should make it more complex than needed, i.e. your proposal is fine.

> Also I propose to name the upcoming release 0.3.0 because in master the (broken) version is 0.2.1. This approach will make obvious that our changes are the most up-to-date.
I fully agree.